### DISCUSSION TITLE
Programming NXP i.MX RT OTP fuse with west

### DISCUSSION THREAD

**Discussion by danie007** [zephyrproject-rtos/zephyr#45521] [General]
Hi all,

We're using NXP's [i.MX RT 1160 EVK](https://www.nxp.com/design/development-boards/i-mx-evaluation-and-development-boards/i-mx-rt1160-evaluation-kit:MIMXRT1160-EVK) and would like to program One Time Programmable (_OTP_) fuses on the processor.

### Currently we can build, sign (_MCUboot_) & flash (_with JTAG_) but the fuse support is unknown

1. Does `west` support signing of eFuses? Is it planned?
2. How can we program fuses on the i.MX RT board?
3. Are there any documentation/issues regarding this?

Thanks in advance.

---

CC @tkmozhi @vishnusudhan

**Comment by marc-hb** [comment]
@danie007 , west itself does not implement building, signing or flashing. These are Zephyr-specific whereas west is a generic wrapper (and multi-repo tool). For instance you can find the code that implements flashing here: https://github.com/zephyrproject-rtos/zephyr/tree/d8b45e38c130/scripts/west_commands/runners

west is merely invoking west "extensions" implemented by Zephyr https://docs.zephyrproject.org/latest/develop/west/build-flash-debug.html

So unless you have any objection I will transfer this issue to https://github.com/zephyrproject-rtos/zephyr/issues

I think details on how you already perform fusing now would be very useful; I mean how is it done without any (future) west indirection.

**Comment by danie007** [comment]
@marc-hb thanks for the overview on `west` and suggesting the appropriate channel for this query. It would be helpful if you could move/transfer this issue to the **Zephyr repo**.

Currently we're using NXP provided tools (_[blhost](https://www.nxp.com/docs/en/user-guide/MCUBLHOSTUG.pdf) & [MFGtool](https://www.nxp.com/docs/en/user-guide/MBOOTMFGTOOLUG.pdf) - pointing to respective user guides_) for programming the OTP fuses on the i.MX RT board.

⬆️
